simson/common/trade.py

```diff
@@ -0,0 +1,59 @@
+import sys
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import model_validator
+from sodym.named_dim_arrays import Parameter
+from typing import Optional, Callable
+
+class Trade(PydanticBaseModel):
+    """ A TradeModule handles the storing and calculation of trade data for a given MFASystem."""
+
+    imports : Parameter
+    exports : Parameter
+    balancer : Optional[Callable] = None
+    predictor : Optional[Callable] = None
+
+
```

Comment:
We could add a validator to check the balancer function:
```suggestion
    @field_validator('balancer')
    @staticmethod
    def check_balancer(v: Callable) -> Callable
        valid_balancers = [balance_by_extremum, balance_by_scaling]
        if v and (v not in valid_balancers):
            raise ValidationError(f"Balancer must be one of {valid_balancers}.")
        return v
```

Reply:
Hmm, out of a coding perspective I find this quite inconvenient as now every time a new balancer is programmed, this now needs to be added to this list, and the validator doesn't check anything functionally, so I'm thinking that it won't help much to catch errors (If I implement a new balancer, I will think it's correct so just automatically add it to the list, which is a bit inconvenient). Though maybe I'm missing something? Also not that passionate either way but this was just my thinking :)

Reply:
Yes, good points. The main advantage is that if the user decides to implement some trade, they would look into the `Trade` class and wonder what the balancer is. This way, they can directly see which functions are implemented that they could use for balancing, and clicking on them, can see how these functions work. This could all be made more explicit if the basic balancer was a class of its own, and all others were made to inherit from it, but maybe this is over-complicating things. Another option, keeping them as functions but giving a new user the idea what the balancer should look like, is just to set a default balancer. And probably the best option (and also the most additional work) would be to have the valid_balancers as a global variable in this file (outside the class), import it into your (currently non-existing) testcases and write a test with @pytest.mark.parametrize and the list of valid_balancers, so that each of them gets checked, also when you add a new one :)